backend/app/models.py

```diff
@@ -36,11 +41,23 @@ class User(db.Model):
     bought_item_id = db.Column(db.Integer, db.ForeignKey('item.id'))
     sold_item = db.relationship('Item', backref='seller', lazy=True, foreign_keys=[sold_item_id])
     bought_item = db.relationship('Item', backref='buyer', lazy=True, foreign_keys=[bought_item_id])
-    roles = db.Column(db.Enum(Role))
+    _roles = db.Column(db.Enum(Role))
 
     def __repr__(self):
         return '<User %r>' % self.username
 
+    @hybrid_property
+    def roles(self):
+        return self._roles
+
+    @roles.setter
+    def roles(self, role):
+        for item in Role:
```

Comment:
Why don't you use `getattr` or `hasattr`? when it's not present raise exception in another case you get the role which you want